staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -27,6 +27,8 @@ const (
 	// More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#label-selector-and-annotation-conventions
 	labelPrefix = "batch.kubernetes.io/"
 
+	JobCreationTimestampAnnotation = labelPrefix + "job-creation-timestamp"
```

Comment:
Make sure to explain in details the purpose of this annotation, its value and expectations user should have wrt to its format and its contents.

Reply:
Added.